control-plane/agents/core/src/core/wrapper.rs

```diff
@@ -217,17 +248,19 @@ impl NodeWrapper {
                     // We failed to fetch all resources from Mayastor so clear all state
                     // information. We take the approach that no information is better than
                     // inconsistent information.
-                    let mut states = self.states.write();
-                    states.clear_all();
+                    self.states.write().clear_all();
+                    self.set_state(NodeStatus::Unknown);
                     Err(e)
                 }
             }
         } else {
             tracing::trace!(
                 "Skipping reload of node '{}' since it's '{:?}'",
                 self.id,
-                self.state
+                self.status
             );
+            let mut states = self.states.write();
+            states.clear_all();
```

Comment:
Nit: You could do `self.states.write().clear_all()` like you do on 251.

Reply:
:+1: